packages/shared/src/components/Attachments/AttachmentItem.tsx

```diff
@@ -0,0 +1,219 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import React from 'react';
+import { Asset, useAssetUpload, useFileInput } from '@frontify/app-bridge';
+import { useSortable } from '@dnd-kit/sortable';
+import {
+    ActionMenu,
+    Button,
+    ButtonEmphasis,
+    Flyout,
+    FlyoutPlacement,
+    IconArrowCircleUp20,
+    IconDocument24,
+    IconGrabHandle20,
+    IconImage24,
+    IconImageStack20,
+    IconPen20,
+    IconPlayFrame24,
+    IconTrashBin20,
+    MenuItemContentSize,
+    MenuItemStyle,
+} from '@frontify/fondue';
+import { forwardRef, useEffect, useState } from 'react';
+import { AttachmentItemProps, SortableAttachmentItemProps } from './types';
+
+const getDecorator = (type: string) => {
+    if (type === 'IMAGE') {
+        return <IconImage24 />;
+    } else if (type === 'VIDEO') {
+        return <IconPlayFrame24 />;
+    } else {
+        return <IconDocument24 />;
+    }
+};
+
+export const AttachmentItem = forwardRef<HTMLDivElement, AttachmentItemProps>(
+    (
+        {
+            item,
+            isEditing,
+            designTokens,
+            draggableProps,
+            transformStyle,
+            isDragging,
+            onAttachmentDelete,
+            onAttachmentReplaceWithBrowse,
+            onAttachmentReplaceWithUpload,
+        },
+        ref
+    ) => {
+        const [selectedAsset, setSelectedAsset] = useState<Asset | undefined>();
+        const [openFileDialog, { selectedFiles }] = useFileInput({ multiple: true, accept: 'image/*' });
+        const [uploadFile, { results: uploadResults, doneAll }] = useAssetUpload();
+
+        useEffect(() => {
+            if (selectedFiles) {
+                uploadFile(selectedFiles[0]);
+            }
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+        }, [selectedFiles]);
+
+        useEffect(() => {
+            if (doneAll) {
+                onAttachmentReplaceWithUpload(uploadResults[0]);
+            }
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+        }, [doneAll, uploadResults]);
+
+        const downloadAttachment = (attachmentUrl: string, filename: string) => {
+            fetch(attachmentUrl).then((response) => {
+                response.blob().then((blob) => {
+                    const url = window.URL.createObjectURL(blob);
```

Comment:
`window.URL.createObjectURL(blob)` could be used directly with
`URL.createObjectURL(blob)`

Reply:
thanks, fixed